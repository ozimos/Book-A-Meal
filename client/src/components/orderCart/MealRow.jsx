import React from 'react';
import PropTypes from 'prop-types';

const MealRow = ({ id, title, price, quantity, setQuantity, removeFromCart }) => {
  const subTotal = price * (quantity || 1);
  return (
    <React.Fragment>
      <tr className="slim">
        <td>{title}</td>
        <td>{price}</td>
        <td>
          <input
            type="number"
            placeholder="1"
            onChange={event => setQuantity(event, id)}
            value={quantity}
            maxLength="4"
          />
        </td>
        <td>{subTotal}</td>
        <td>
          <button className="btn btn-cart" onClick={() => removeFromCart(id)}>
                &#10006;
          </button>
        </td>
      </tr>
    </React.Fragment>
  );
};

MealRow.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  removeFromCart: PropTypes.func.isRequired,
  setQuantity: PropTypes.func.isRequired,
};
export default MealRow;
