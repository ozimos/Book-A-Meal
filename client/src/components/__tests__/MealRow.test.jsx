/* global React:false, shallow:false */
/* eslint react/jsx-indent: off */

import { MemoryRouter } from 'react-router-dom';
import MealRow from '../orderCart/MealRow';

describe('Login Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}><MealRow />
                            </MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });
});

