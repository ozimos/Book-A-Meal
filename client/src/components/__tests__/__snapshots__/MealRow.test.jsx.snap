// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealRow Component renders correctly 1`] = `
<React.Fragment>
  <tr
    className="slim"
  >
    <td>
      abc
    </td>
    <td>
      4
    </td>
    <td>
      <input
        maxLength="4"
        onChange={[Function]}
        placeholder="1"
        type="number"
        value={4}
      />
    </td>
    <td>
      16
    </td>
    <td>
      <button
        className="btn btn-cart"
        onClick={[Function]}
      >
        âœ–
      </button>
    </td>
  </tr>
</React.Fragment>
`;
